### DISCUSSION TITLE
[Proposal] fluent interface version of List<T>.ForEach(Action<T>)

### DISCUSSION THREAD

**Discussion by NagayamaToshiaki** [dotnet/csharplang#4943] [Language Ideas]
It should be neat if we can chain the array definition and ForEach iteration like:
`var list = db.Foo.Where(f =>f.bar = "bar" ).ForEach(f => f.baz += 10)`
I knew "Modifying the underlying collection in the body of the Action<T> delegate is not supported" but our team actually using modification by ForEach method.

**Comment by Joe4evr** [comment]
https://ericlippert.com/2009/05/18/foreach-vs-foreach/

**Reply by 333fred** [reply]
This would also be a request for dotnet/runtime, not this repository.
